plumbing/object/change.go

```diff
@@ -61,6 +70,12 @@ func (c *Change) Files() (from, to *File, err error) {
 	return
 }
 
+func isInvalidFileMode(t TreeEntry) bool {
```

Comment:
@mcuadros I think `FileMode.IsRegular()` doesn't cover all the cases, like executable files. Maybe @alcortesm could do us some hints.